src/commands/object/sign.ts

```diff
@@ -39,21 +39,29 @@ export class SignCommand extends Command{
     }
     protected async onRun(params:signOptions):Promise<Command>{
         const session = get_session();
+
         let alg: graphene.MechanismType;
-        if(!params.mech){
-            console.log("No mechanism found. Defaulting to ECDSA.");
-            alg = graphene.MechanismEnum.ECDSA;
-        }else{
-            alg = params.mech;
+
+        let foundIndex = 0;
+        for(let i=268;i<Object.keys(graphene.MechanismEnum).length;i++){
```

Comment:
Would something like this work here?

https://github.com/ictinusinc/graphene-cli/blob/0c88c66b154462c90eaf0af6cf5a13112d858c8f/src/commands/slot/info.ts#L28-L34